libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1075,19 +1075,24 @@ void AP_InertialSensor::wait_for_sample(void)
     if (!_hil_mode) {
         // we also wait for at least one backend to have a sample of both
         // accel and gyro. This normally completes immediately.
-        bool gyro_available = false;
-        bool accel_available = false;
-        while (!gyro_available || !accel_available) {
+        while (true) {
+            bool gyro_available = false;
+            bool accel_available = false;
```

Comment:
These are not the same thing because you are resetting both to false here. The intention is to exit the loop when we have both set, but they won't necessarily be set on the same run of the loop